include/barcodes/datamatrix.php

```diff
@@ -778,7 +778,7 @@ protected function getHighLevelEncoding($data) {
 							if (isset($this->chset['SH1'][$chr])) {
 								$temp_cw[] = 0; // shift 1
 								$shiftset = $this->chset['SH1'];
-							} elseif (isset($chr, $this->chset['SH2'][$chr])) {
+							} elseif (isset($this->chset['SH2'][$chr])) {
```

Comment:
Note this code here has several bugs and it is way behind:
https://github.com/tecnickcom/tc-lib-barcode
It would be better to switch to the new library.

Reply:
`$chr` is always set, that is all